### PR TITLE
Fix compilation warnings

### PR DESCRIPTION
**Pull request by noonfom**
### Summary of changes 
Recently, a user found that compiling Mbed OS with the GCC_ARM toolchain produces several warnings that could affect program reliability (ARMmbed#13604). It is not possible to fix all the warnings since there are a lot of 3rd party components in the tree. However, simple solutions exist for a some of the warnings. A few are given in ARMmbed/mbedtls#3867; others are supplied here. 

compout.txt gives the compiler output after making the changes.

`mbed compile -c -m NUCLEO_F446RE -t GCC_ARM &> ../compout.txt`

[compout.txt](https://github.com/noonfom/mbed-os/files/5517501/compout.txt)

warning.txt gives the remaining warnings. 
 
`cat compout.txt | grep "Warning" > warning.txt`

[warning.txt](https://github.com/noonfom/mbed-os/files/5517510/warning.txt)

 
      
### Documentation 
None

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [ ] Feature update (New feature / Functionality change / New API)
    [ ] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 

    [ ] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [ ] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers 
@pan- 
@paul-szczepanek-arm

----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: connectivity/nanostack/coap-service/source/coap_connection_handler.c [hunk 1]
```diff
@@ -237,7 +237,9 @@ static void clear_secure_sessions(internal_socket_t *this)
     if (this) {
         ns_list_foreach_safe(secure_session_t, cur_ptr, &secure_session_list) {
             if (cur_ptr->parent == this) {
-                coap_security_send_close_alert(cur_ptr->sec_handler);
+                #if COAP_SECURITY_AVAILABLE
```
Comment (paul-szczepanek-arm) [inline connectivity/nanostack/coap-service/source/coap_connection_handler.c:240]: align #if to 0 and the line below should not be undented

**Review by paul-szczepanek-arm — CHANGES_REQUESTED** [review]
Thanks. Please fix indent and add the #ifdefs around the unusued functions:
```[Warning] coap_connection_handler.c@544,12: 'timer_status' defined but not used [-Wunused-function]
[Warning] coap_connection_handler.c@521,13: 'start_timer' defined but not used [-Wunused-function]
[Warning] coap_connection_handler.c@468,12: 'secure_session_recvfrom' defined but not used [-Wunused-function]
[Warning] coap_connection_handler.c@435,12: 'secure_session_sendto' defined but not used [-Wunused-function]```

File: connectivity/nanostack/coap-service/source/coap_connection_handler.c [hunk 2]
```diff
@@ -237,7 +237,9 @@ static void clear_secure_sessions(internal_socket_t *this)
...
+                    coap_security_send_close_alert(cur_ptr->sec_handler);
+                #endif //COAP_SECURITY_AVAILABLE
```
Comment (paul-szczepanek-arm) [inline connectivity/nanostack/coap-service/source/coap_connection_handler.c:242]:
```suggestion
#if COAP_SECURITY_AVAILABLE
                coap_security_send_close_alert(cur_ptr->sec_handler);
#endif //COAP_SECURITY_AVAILABLE
```

Comment (paul-szczepanek-arm) [reply connectivity/nanostack/coap-service/source/coap_connection_handler.c:242]: same below

Comment (noonfom) [reply connectivity/nanostack/coap-service/source/coap_connection_handler.c]: Thanks. This has been resolved in commit 9a18366ad9517604fc32354a0357e136be189df7.

**Comment by paul-szczepanek-arm** [comment]
should be [x] Covered by existing mbed-os tests (Greentea or Unittest)

**Review by pan- — COMMENTED** [review]
Thanks @noonfom that looks good from my perspective however we have to sync with the Oulu team to understand how they want to upstream changes to the nanostack. 

Can you open a PR against mbed-os ?

**Pull request closed without merging**